etp-backend/src/main/clj/solita/etp/service/energiatodistus_search.clj

```diff
@@ -117,21 +118,28 @@
     #(expression-seq->sql "and" predicate-expression->sql %)
     where))
 
-(def blank? (some-fn nil? empty?))
+(defn whoami->sql [{:keys [id] :as whoami}]
+  (cond
+    (rooli-service/paakayttaja? whoami) ["energiatodistus.tila_id <> ?" 0]
+    (rooli-service/laatija? whoami) ["energiatodistus.laatija_id = ?" id]))
 
-(defn query->sql [{:keys [where sort order limit offset]}]
+(defn sql-query [whoami {:keys [where sort order limit offset]}]
   (schema/validate [[[(schema/one schema/Str "predicate") schema/Any]]] where)
-
-  (let [[where-sql & params] (where->sql where)]
-    (cons (str base-query
-               (when-not (blank? where-sql) (str \newline "where " where-sql))
-               (when-not (blank? sort)
-                 (str \newline "order by " (field->sql sort) " " (or order "asc")))
-               (str \newline "limit " (or limit 100))
-               (when-not (nil? offset) (str \newline "offset " offset)))
-          params)))
-
-(defn search [db query]
+  (let [[where-sql & where-params] (where->sql where)
+        [visibility-sql & visibility-params] (whoami->sql whoami)]
+    (concat [(str base-query
+                  \newline
+                  "where "
+                  visibility-sql
+                  (when-not (str/blank? where-sql) (str " and " where-sql))
```

Comment:
where-sql:n pitää luultavasti olla sulkeissa koska and:n precedence on suurempi kuin or:in